controllers/space/space_controller_test.go

```diff
@@ -594,6 +595,274 @@ func TestDeleteSpace(t *testing.T) {
 	})
 }
 
+func TestUpdate(t *testing.T) {
+
+	// given
+	logf.SetLogger(zap.New(zap.UseDevMode(true)))
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+	basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates)
+	otherTier := tiertest.OtherTier()
+
+	t.Run("update needed", func(t *testing.T) {
+		// given that Space is promoted to `other` tier and corresponding NSTemplateSet is not up-to-date
+		s := spacetest.NewSpace("oddity",
+			spacetest.WithTierNameFor(otherTier), // assume that at this point, the `TemplateTierHash` label was already removed by the ChangeTierRequestController
+			spacetest.WithTargetCluster("member-1"),
+			spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+			spacetest.WithFinalizer())
+		hostClient := test.NewFakeClient(t, s, basicTier, otherTier)
+		nstmplSet := nstemplatetsettest.NewNSTemplateSet("oddity", nstemplatetsettest.WithReadyCondition())
+		member1Client := test.NewFakeClient(t, nstmplSet)
+		member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+		member2 := NewMemberCluster(t, "member-2", corev1.ConditionTrue)
+		ctrl := newReconciler(hostClient, member1, member2)
+
+		// when
+		res, err := ctrl.Reconcile(context.TODO(), requestFor(s))
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: 3 * time.Second,
+		}, res) // explicitly requeue while the NSTemplate update is triggered by its controller
+		spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
+			Exists().
+			HasTier(otherTier.Name).
+			HasTargetCluster("member-1").
+			HasStatusTargetCluster("member-1").
+			HasConditions(spacetest.Provisioning()).
+			DoesNotHaveLabel(tierutil.TemplateTierHashLabelKey(otherTier.Name)) // not set yet, since NSTemplateSet must be updated first
+		nsTmplSet := nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, "oddity", member1.Client).
+			Exists().
+			HasTierName(otherTier.Name).
+			Get()
+
+		t.Run("requeue while NSTemplateSet is not ready", func(t *testing.T) {
+			// given another round of requeue without while NSTemplateSet is *not ready*
+			nsTmplSet.Status.Conditions = []toolchainv1alpha1.Condition{
+				nstemplatetsettest.Provisioning(),
+			}
+			err := member1.Client.Update(context.TODO(), nsTmplSet)
+			require.NoError(t, err)
+			ctrl := newReconciler(hostClient, member1, member2)
```

Comment:
it's the same thing as last time - we don't need to create the reconcile again when the client and controller is supposed to be reused in the sub-test
```suggestion
```